src/PrestaShopBundle/Twig/Layout/PrestaShopLayoutGlobalVariables.php

```diff
@@ -28,168 +28,104 @@
 
 namespace PrestaShopBundle\Twig\Layout;
 
-use PrestaShop\PrestaShop\Adapter\Configuration;
 use PrestaShop\PrestaShop\Adapter\LegacyContext;
-use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagStateCheckerInterface;
-use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
-use PrestaShop\PrestaShop\Core\Localization\Locale\Repository as LocaleRepository;
-use PrestaShopBundle\Entity\Repository\TabRepository;
-use PrestaShopBundle\Entity\Tab;
-use PrestaShopBundle\Service\DataProvider\UserProvider;
-use Shop;
-use Symfony\Component\HttpFoundation\RequestStack;
-use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
-use Tools;
 
+/**
+ * Allows you to define variables accessible globally in a twig rendering.
+ * Only public methods will be accessible on the rendering.
+ */
 class PrestaShopLayoutGlobalVariables
```

Comment:
I wonder if it's really relevant to have `PrestaShopLayoutGlobalVariables` and `TemplateVariables` services, this one basically returns all the getters from `TemplateVariables` no? The only addition is the `setupSmarty` method 

But except for this method that, hopefully, we'll improve/remove at some point wouldn't it make sense to merge both services and keep only one?

Reply:
From my point of view, I don't think so. The TemplateVariables service was added to address the code duplication that exists everywhere in the Symfony layout. There was going to be even more with the addition of the Smarty context.

There is variables in TemplateVariables that is not found in the global layout.